.pre-commit-config.yaml

```diff
@@ -1,33 +1,77 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+
+minimum_pre_commit_version: '2.8.2'
+
+default_language_version:
+  python: python3
+
+default_stages: [commit]
+
+
 repos:
+  # General file checks and fixers
   - repo: https://github.com/pre-commit/pre-commit-hooks
-    rev: v3.4.0
+    rev: v4.1.0
     hooks:
       - id: mixed-line-ending
-        name: Normalize mixed line endings
+        name: "Normalize mixed line endings"
         args: [--fix=lf]
+      - id: fix-byte-order-marker
+        name: "Remove Unicode BOM"
+      - id: file-contents-sorter
+        name: "Sort codespell ignore list"
+        files: '.codespell/ignore-words.txt'
+
+      - id: check-case-conflict
+        name: "Check for case conflicts"
+      - id: check-merge-conflict
+        name: "Check for merge conflict markers"
+      - id: check-executables-have-shebangs
+        name: "Check that executables have shebangs"
+      - id: check-shebang-scripts-are-executable
+        name: "Check that shebangs are executable"
+
+      - id: check-vcs-permalinks
+        name: "Check that VCS links are permalinks"
 
+      - id: check-ast
+        name: "Check Python AST"
+      - id: check-json
+        name: "Check JSON"
+      - id: check-toml
+        name: "Check TOML"
```

Comment:
Noting there's no TOML files in the repo [yet](https://python.github.io/peps/pep-0680/), but this won't run anyway if no TOML files aren't found. 👍